src/accelerate/accelerator.py

```diff
@@ -1258,9 +1258,11 @@ def prepare_model(self, model: torch.nn.Module, device_placement=None):
             model, "hf_device_map", False
         ):
             model_devices = set(model.hf_device_map.values())
-            if len(model_devices) > 1:
+            if len(model_devices) > 1 and self.distributed_type == DistributedType.MULTI_GPU:
```

Comment:
```suggestion
            if len(model_devices) > 1 and not (self.distributed_type != DistributedType.NO):
```
maybe this is a better check

Reply:
Yes this is a better check, but you need to also test it is not the (deprecated) `DistributedType.MPS` (although maybe we can remove this one now @muellerzr ?)

Reply:
I see, I think let's add this suggestion as it also includes that the distributed type is not `DistributedType.MPS`